### ISSUE TITLE
[Bug]: mathtext not always rendering combining accents in the same way

### ISSUE THREAD

**Issue by oscargus** [matplotlib/matplotlib#23257]
### Bug summary

In the reference figure https://github.com/matplotlib/matplotlib/blob/main/lib/matplotlib/tests/baseline_images/test_mathtext/mathtext_dejavuserif_00.pdf

the dot above the s is differently positioned. Not clear why as the expressions are `\dot s` and `\dot{s}` respectively. This doesn't happen to all fonts (maybe only this?).

It also happens in the svg image, although not as clear. Looking at the source, the x-position difference between the accent and character is 54.4 for the left and 57.5 for the right.

### Code for reproduction

```python
import matplotlib.pyplot
plt.text(0.5, 0.5, r'$\dot s + \dot{s}$')
plt.savefig('sdot.pdf')
plt.savefig('sdot.svg')
```

### Actual outcome

Different rendering

### Expected outcome

Identical rendering

### Matplotlib Version

3.6.0.dev2642+g25292618a9.d20220612

### Installation

git checkout

**Comment by tfpf** [comment]
It looks like this happens with all fonts. I don't think it's specific to the SVG and PDF backends, since the differences in rendering are visible in the plot window. (I am using the GTK3Agg backend.)
```
import matplotlib.pyplot as plt
fig = plt.figure()
for (i, fontset) in zip([.1, .3, .5, .7, .9], ['cm', 'dejavusans', 'dejavuserif', 'stix', 'stixsans']):
    plt.rc('mathtext', fontset=fontset)
    fig.text(i, 0.5, r'$\dot s$', ha='center', va='center', size=250, alpha=0.5, color='red')
    fig.text(i, 0.5, r'$\dot{s}$', ha='center', va='center', size=250, alpha=0.5, color='green')
plt.show()
```
![image](https://user-images.githubusercontent.com/19171016/177809386-2b1a6907-861a-44ca-bba9-d84e7fa0885a.png)

The root of the problem is the difference in the way `s` and `{s}` are treated. The former is a symbol (which gets converted into a `Char`), whereas the latter is a group (which gets inserted into an `Hlist`).
https://github.com/matplotlib/matplotlib/blob/c936ef5cf0713e1f3a56614db5eccd5d32fe3632/lib/matplotlib/_mathtext.py#L2122-L2124

By default, an `Hlist` adds kerning to characters, so `{s}` is actually rendered with some additional space after it. The dot is then centred above the `Hlist`. That's why it is displaced to the right.

**Comment by anntzer** [comment]
Good catch.  The behavior of `\dot{s}` would seem to be in error, no?

**Comment by oscargus** [comment]
I think that it is an error, yes. At least https://www.ctan.org/pkg/comprehensive writes with {} around the character for accents. (And I always was under the impression that it is the same, but indeed a less authoritative source...)

Indeed a good catch @tfpf !

**Comment by tfpf** [comment]
I agree with oscargus. LaTeX does not add that extra space at the end, so the behaviour `\dot{s}` is incorrect. However, LaTeX's behaviour resembles that of `\dot{s}`, probably because it shifts the accent when the character is italicised. Which is might be the purpose of `sym.width / 4.0` in the `accent` function.

https://github.com/matplotlib/matplotlib/blob/c936ef5cf0713e1f3a56614db5eccd5d32fe3632/lib/matplotlib/_mathtext.py#L2067

In this case, the interplay of the italic correction and the extra space at the end mess up the placement.

I should add that the point about the italic correction has already been noted in https://github.com/matplotlib/matplotlib/issues/19299.

**Comment by oscargus** [comment]
I think that #23189 actually solves this (and explains one of the failing tests). One can probably make a simpler patch that just handles this, something like:

```python
             if not isinstance(sym, Char):
                sym = sym.children[0]
```
if it is expected that #23189 will take a long time to get ready (I think that it is probably worthwhile to spend the effort there if we are still planning to "break" how the accents are rendered...).

**Comment by anntzer** [comment]
Looks like something worth revisiting now that raqm is in, as we could perhaps now render all (nonwide) accents as combining characters.  See also some early work on this at #23189.

Tagging as part of text-overhaul (as it logically belongs together), but I don't think it's worth blocking the merge of text-overhaul into main with this -- changing this will only break the test images *that contain accents*, which should be relatively few.
